Emby.Server.Implementations/AppBase/BaseConfigurationManager.cs

```diff
@@ -171,16 +171,29 @@ public virtual void ReplaceConfiguration(BaseApplicationConfiguration newConfigu
         private void UpdateCachePath()
         {
             string cachePath;
-
+            // If the configuration file has no entry (i.e. not set in UI)
             if (string.IsNullOrWhiteSpace(CommonConfiguration.CachePath))
             {
-                cachePath = null;
+                // If the current live configuration has no entry (i.e. not set on CLI/envvars, during startup)
+                if (string.IsNullOrWhiteSpace(((BaseApplicationPaths)CommonApplicationPaths).CachePath))
+                {
+                    // Set cachePath to a default value under ProgramDataPath
+                    cachePath = (((BaseApplicationPaths)CommonApplicationPaths).ProgramDataPath + "/cache");
```

Comment:
Generally we use Path.Combine() for this to allow for windows path separators, so your line becomes:

`Path.Combine(((BaseApplicationPaths)CommonApplicationPaths).ProgramDataPath, "cache");`

Reply:
I was just thinking about that but wasn't sure exactly how - thanks!